src/components/AudioPlayer/AudioPlayer.tsx

```diff
@@ -0,0 +1,192 @@
+   */
+  filePath?: string;
+  /** Optional string to set the text for a `Heading` component */
+  headingText?: string;
+  /** Optional string to set the text for a `HelperErrorText` component */
+  helperText?: string;
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
+  /** Optional title to added to the `<iframe>` element for improved accessibility. If omitted, a
+   * generic title will be added
+   */
+  iframeTitle?: string;
+}
+
+/**
+ * Renders an audio player from 3rd party services such as Libsyn, SoundCloud, and Spotify.
+ */
+export const AudioPlayer = chakra(
+  forwardRef<HTMLDivElement, React.PropsWithChildren<AudioPlayerProps>>(
+    (props, ref?) => {
+      const {
+        audioType,
+        className,
+        descriptionText,
+        embedCode,
+        filePath,
+        headingText,
+        helperText,
+        id,
+        iframeTitle = null,
+        ...rest
+      } = props;
+
+      // The root iframe object generated from the embedCode,
+      const iframeDoc = parseIframeEmbedCode(embedCode);
+      // when no embedCode or it was a broken code
+      let isInvalidEmbed =
+        !embedCode || !iframeDoc || !isValidEmbedCode(iframeDoc);
+
+      const errorMessage =
+        "<strong>Error: </strong>This audio player has not been configured properly. Please contact the site administrator.";
+
+      function isValidEmbedCode(doc: HTMLIFrameElement): boolean {
+        return audioType !== "file" && doc?.src?.includes(audioType);
+      }
+
+      if (iframeTitle && audioType === "file") {
+        console.warn(
+          'NYPL Reservoir AudioPlayer: The `iframeTitle` prop will be ignored when the `audioType` prop is set to "file."'
+        );
+      }
+      // Only set the iframe title if it doesn't already have it in the iframe
+      if (iframeDoc && !iframeDoc.title) {
+        iframeDoc.title = iframeTitle ? iframeTitle : "Embedded audio player";
+      }
+
+      // A friendly message states that we don't support filePath prop
+      const isLocalAudioFile = filePath || audioType === "file";
+      if (isLocalAudioFile) {
+        console.warn(
+          "The current AudioPlayer does not support locally hosted audio file."
+        );
+      }
+
+      const isLocalWithEmbed = audioType === "file" && embedCode;
+      if (isLocalWithEmbed) {
+        // Question: should we show this now?
+        console.warn(
+          'NYPL Reservoir AudioPlayer: The `embedCode` prop will be ignored when the audioType prop is set to "file".'
+        );
+      }
+      const isLocalWithoutPath = audioType === "file" && !filePath;
+      if (isLocalWithoutPath) {
+        // Question: Should we remove this warning since we don't support filepath?
+        console.warn(
+          'NYPL Reservoir AudioPlayer: The `filePath` prop is required when the `audioType` prop is set to "file."'
+        );
+      }
```

Comment:
Yeah, I'm not sure if we really need this now and maybe the first message above is enough.